### ISSUE TITLE
Returns Empty JSON in Multi-turn Dialogue Script

### ISSUE THREAD

**Issue by da13132** [IDEA-Research/Rex-Omni#17]
Thank you very much for your excellent open-source work.

When I'm running a contextual conversation to perform inference on a pose estimation task, the model only outputs a **JSON file with empty values** for all coordinates. However, when using the official inference code, keypoint inference works perfectly. I haven't found the reason yet. I don't know if you have encountered this situation. I hope to get your help.

#### 1. Prompt Used
My prompt matches the official one:
> Can you detect each person in the image using a `[x0, y0, x1, y1]` box format, and then provide the coordinates of its nose, left eye, right eye, left ear, right ear, left shoulder, right shoulder, left elbow, right elbow, left wrist, right wrist, left hip, right hip, left knee, right knee, left ankle, right ankle as `[x0, y0]`? Output the answer in JSON format.

#### 2. Actual Output
The output of the model under this prompt is:

```shell
👤 User: Can you detect each person in the image using a [x0, y0, x1, y1] box format, and then provide the coordinates of its nose, left eye, right eye, left ear, right ear, left shoulder, right shoulder, left elbow, right elbow, left wrist, right wrist, left hip, right hip, left knee, right knee, left ankle, right ankle as [x0, y0]? Output the answer in JSON format.
🤔 Thinking...
Processed prompts: 100%|███████████████| 1/1 [00:05<00:00, 5.07s/it, est. speed input: 310.20 toks/s, output: 92.04 toks/s]

🤖 Assistant: ```json
{"person1": {"bbox": "", "keypoints": {"nose": "", "left eye": "", "right eye": "", "left ear": "", "right ear": "", "left shoulder": "", "right shoulder": "", "left elbow": "", "right elbow": "", "left wrist": "", "right wrist": "", "left hip": "", "right hip": "", "left knee": "", "right knee": "", "left ankle": "", "right ankle": ""}}, "person2": {"bbox": "", "keypoints": {"nose": "", "left eye": "", "right eye": "", "left ear": "", "right ear": "", "left shoulder": "", "right shoulder": "", "left elbow": "", "right elbow": "", "left wrist": "", "right wrist": "", "left hip": "", "right hip": "", "left knee": "", "right knee": "", "left ankle": "", "right ankle": ""}}, "person3": {"bbox": "", "keypoints": {"nose": "", "left eye": "", "right eye": "", "left ear": "", "right ear": "", "left shoulder": "", "right shoulder": "", "left elbow": "", "right elbow": "", "left wrist": "", "right wrist": "", "left hip": "", "right hip": "", "left knee": "", "right knee": "", "left ankle": "", "right ankle": ""}}}
```
Here is my multi-round dialogue code that is causing the issue:
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Rex-Omni Multi-GPU continuous dialogue script
Supports both Transformers and VLLM backends
"""

import os
import json
import time
import torch
import argparse
from PIL import Image
from typing import Optional, Dict, Any, List
from qwen_vl_utils import process_vision_info, smart_resize


class ChatBot:
    """Multi-GPU ChatBot (supports dual backends)"""
    
    def __init__(self, model_path: str, backend: str = "transformers", **kwargs):
        """
        Initializes the ChatBot.
        
        Args:
            model_path: Path to the model.
            backend: "transformers" or "vllm".
            **kwargs: Other parameters (temperature, top_p, max_tokens, etc.).
        """
        self.model_path = model_path
        self.backend = backend.lower()
        
        # Generation parameters
        self.temperature = kwargs.get('temperature', 0.0)
        self.top_p = kwargs.get('top_p', 0.05)
        self.top_k = kwargs.get('top_k', 1)
        self.max_tokens = kwargs.get('max_tokens', 2048)
        self.repetition_penalty = kwargs.get('repetition_penalty', 1.05)
        
        # Image processing parameters
        self.system_prompt = kwargs.get('system_prompt', 'You are a helpful assistant')
        self.min_pixels = kwargs.get('min_pixels', 16 * 28 * 28)
        self.max_pixels = kwargs.get('max_pixels', 2560 * 28 * 28)
        
        # Initialize conversation history (with system message)
        self.conversation_history = [
            {"role": "system", "content": self.system_prompt}
        ]
        
        self._init_model(**kwargs)
        print("✅ Model loaded successfully!\n")
    
    def _init_model(self, **kwargs):
        """Initializes the model."""
        print(f"🚀 Loading model ({self.backend})...")
        
        if self.backend == "transformers":
            from transformers import Qwen2_5_VLForConditionalGeneration, AutoProcessor
            
            dtype = torch.float16 if kwargs.get('dtype') == 'float16' else torch.bfloat16
            
            self.processor = AutoProcessor.from_pretrained(
                self.model_path, trust_remote_code=True
            )
            # Standardize on left padding and ensure a pad_token exists
            try:
                self.processor.tokenizer.padding_side = "left"
                if self.processor.tokenizer.pad_token_id is None:
                    self.processor.tokenizer.pad_token = self.processor.tokenizer.eos_token
            except Exception:
                pass
            self.model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
                self.model_path,
                torch_dtype=dtype,
                device_map=kwargs.get('device_map', 'auto'),
                trust_remote_code=True,
            )
            # Explicitly align generation config
            try:
                if getattr(self.model, 'generation_config', None) is not None:
                    if self.model.generation_config.pad_token_id is None and self.processor.tokenizer.pad_token_id is not None:
                        self.model.generation_config.pad_token_id = self.processor.tokenizer.pad_token_id
                    if self.model.generation_config.eos_token_id is None and self.processor.tokenizer.eos_token_id is not None:
                        self.model.generation_config.eos_token_id = self.processor.tokenizer.eos_token_id
            except Exception:
                pass
            
        elif self.backend == "vllm":
            from transformers import AutoProcessor
            from vllm import LLM, SamplingParams
            
            self.processor = AutoProcessor.from_pretrained(
                self.model_path, trust_remote_code=True
            )
            self.processor.tokenizer.padding_side = "left"
            
            # Automatically infer tensor parallel size
            tp_size = kwargs.get('tensor_parallel_size', 1)
            try:
                if tp_size is None or tp_size <= 0:
                    visible = os.environ.get("CUDA_VISIBLE_DEVICES")
                    if visible and visible.strip():
                        tp_size = len([d for d in visible.split(',') if d.strip() != ''])
                    else:
                        tp_size = torch.cuda.device_count() if torch.cuda.is_available() else 1
            except Exception:
                tp_size = 1
            print(f"⚙️  VLLM tensor parallel size: {tp_size}")

            self.model = LLM(
                model=self.model_path,
                tokenizer=self.model_path,
                tokenizer_mode="slow",
                max_model_len=kwargs.get('max_model_len', 4096),
                gpu_memory_utilization=kwargs.get('gpu_memory_utilization', 0.9),
                tensor_parallel_size=tp_size,
                trust_remote_code=True,
            )
            
            self.sampling_params = SamplingParams(
                max_tokens=self.max_tokens,
                temperature=self.temperature,
                top_p=self.top_p,
                top_k=self.top_k,
                repetition_penalty=self.repetition_penalty,
                stop=["<|im_end|>"],
            )
    
    def chat(self, text: str, image_path: Optional[str] = None) -> str:
        """
        Chat interface.
        
        Args:
            text: User input text.
            image_path: Path to the image (optional).
            
        Returns:
            Model's response.
        """
        # Build the message
        if image_path:
            try:
                image = Image.open(image_path).convert("RGB")
                
                # Calculate resized parameters (needed for tasks like keypoint detection)
                if self.backend == "transformers":
                    w, h = image.size
                    resized_height, resized_width = smart_resize(
                        h, w, 28,
                        min_pixels=self.min_pixels,
                        max_pixels=self.max_pixels
                    )
                    content = [
                        {
                            "type": "image",
                            "image": image,
                            "resized_height": resized_height,
                            "resized_width": resized_width,
                        },
                        {"type": "text", "text": text}
                    ]
                else:  # vllm
                    content = [
                        {
                            "type": "image",
                            "image": image,
                            "min_pixels": self.min_pixels,
                            "max_pixels": self.max_pixels,
                        },
                        {"type": "text", "text": text}
                    ]
                print(f"🖼️  Image loaded: {image_path}")
            except:
                content = text
        else:
            content = text
        
        self.conversation_history.append({"role": "user", "content": content})
        
        # Generate response
        if self.backend == "transformers":
            response = self._generate_transformers()
        else:
            response = self._generate_vllm()
        
        self.conversation_history.append({"role": "assistant", "content": response})
        return response
    
    def _generate_transformers(self) -> str:
        """Transformers backend generation."""
        from qwen_vl_utils import process_vision_info
        
        text = self.processor.apply_chat_template(
            self.conversation_history, tokenize=False, add_generation_prompt=True
        )
        
        # Extract image list
        image_list = []
        for msg in self.conversation_history:
            if msg["role"] == "user" and isinstance(msg.get("content"), list):
                for item in msg["content"]:
                    if item.get("type") == "image":
                        image_list.append(item["image"])
        
        inputs = self.processor(
            text=[text],
            images=image_list if image_list else None,
            padding=True,
            return_tensors="pt"
        )
        
        # Move to device for single-card inference; for device_map, HF handles placement
        device = getattr(self.model, 'device', None)
        if device is not None and str(device) != 'meta':
            inputs = inputs.to(device)
        
        with torch.no_grad():
            generated_ids = self.model.generate(
                **inputs,
                max_new_tokens=self.max_tokens,
                temperature=self.temperature,
                top_p=self.top_p,
                top_k=self.top_k,
                repetition_penalty=self.repetition_penalty,
                do_sample=self.temperature > 0,
                pad_token_id=self.processor.tokenizer.pad_token_id or self.processor.tokenizer.eos_token_id,
                eos_token_id=self.processor.tokenizer.eos_token_id,
            )
        
        generated_ids_trimmed = [
            out_ids[len(in_ids):] 
            for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
        ]
        
        return self.processor.batch_decode(
            generated_ids_trimmed, skip_special_tokens=False, clean_up_tokenization_spaces=False
        )[0]
    
    def _generate_vllm(self) -> str:
        """VLLM backend generation."""
        from qwen_vl_utils import process_vision_info
        
        text = self.processor.apply_chat_template(
            self.conversation_history, tokenize=False, add_generation_prompt=True
        )
        
        image_inputs, video_inputs = process_vision_info(self.conversation_history)
        
        mm_data = {}
        if image_inputs:
            mm_data["image"] = image_inputs
        if video_inputs:
            mm_data["video"] = video_inputs
        
        outputs = self.model.generate(
            [{"prompt": text, "multi_modal_data": mm_data}],
            sampling_params=self.sampling_params
        )
        
        return outputs[0].outputs[0].text.strip()
    
    def clear_history(self):
        """Clears the conversation history (keeps the system message)."""
        self.conversation_history = [
            {"role": "system", "content": self.system_prompt}
        ]
        print("🗑️  Conversation history cleared.")
    
    def show_history(self):
        """Displays the conversation history."""
        if len(self.conversation_history) <= 1:  # Only system message
            print("📝 Conversation history is empty.")
            return
        
        print("\n" + "="*60)
        print("📝 Conversation History:")
        print("="*60)
        for i, msg in enumerate(self.conversation_history[1:], 1):  # Skip system message
            role = "👤 User" if msg["role"] == "user" else "🤖 Assistant"
            content = msg["content"]
            if isinstance(content, str):
                print(f"{i}. {role}: {content}")
            else:
                text_parts = [c["text"] for c in content if c.get("type") == "text"]
                has_image = any(c.get("type") == "image" for c in content)
                image_mark = " [🖼️]" if has_image else ""
                print(f"{i}. {role}{image_mark}: {' '.join(text_parts)}")
        print("="*60 + "\n")


def interactive_mode(chatbot: ChatBot):
    """Interactive mode."""
    print("\n" + "="*60)
    print("🎉 Rex-Omni Multi-GPU Dialogue System")
    print("="*60)
    print("Commands: /img <path> - Load an image")
    print("          /clear     - Clear history")
    print("          /history   - Show history")
    print("          /quit      - Exit\n")
    
    pending_image = None
    
    while True:
        try:
            prompt = "👤 User [🖼️]: " if pending_image else "👤 User: "
            user_input = input(prompt).strip()
            
            if not user_input:
                continue
            
            if user_input.lower() in ['/quit', '/exit']:
                print("\n👋 Goodbye!")
                break
            
            elif user_input.lower() == '/clear':
                chatbot.clear_history()
                pending_image = None
                continue
            
            elif user_input.lower() == '/history':
                chatbot.show_history()
                continue
            
            elif user_input.lower().startswith('/img '):
                image_path = user_input.split(maxsplit=1)[1].strip()
                if os.path.exists(image_path):
                    pending_image = image_path
                    print(f"✅ Image staged: {image_path}")
                else:
                    print(f"❌ Image not found: {image_path}")
                continue
            
            # Generate response
            print("🤔 Thinking...")
            response = chatbot.chat(user_input, pending_image)
            print(f"\n🤖 Assistant: {response}\n")
            
            pending_image = None
            
        except KeyboardInterrupt:
            print("\n\n👋 Goodbye!")
            break
        except Exception as e:
            print(f"\n❌ Error: {e}\n")


def main():
    parser = argparse.ArgumentParser(description="Rex-Omni Multi-GPU Dialogue")
    parser.add_argument('--model_path', type=str, 
                        default='/home/zl111111/Rex-Omni/qwenpose/Rex-Omni',
                        help='Path to the model')
    parser.add_argument('--backend', type=str, default='transformers',
                        choices=['transformers', 'vllm'], help='Inference backend')
    parser.add_argument('--temperature', type=float, default=0.0, help='Sampling temperature')
    parser.add_argument('--max_tokens', type=int, default=2048, help='Max new tokens to generate')
    parser.add_argument('--top_p', type=float, default=0.05, help='Top-p (nucleus) sampling threshold')
    parser.add_argument('--top_k', type=int, default=1, help='Top-k sampling')
    parser.add_argument('--repetition_penalty', type=float, default=1.05, help='Repetition penalty')
    
    # Transformers args
    parser.add_argument('--dtype', type=str, default='bfloat16',
                        choices=['float16', 'bfloat16'], help='Data type')
    parser.add_argument('--device_map', type=str, default='auto', help='Device map')
    
    # VLLM args
    parser.add_argument('--tensor_parallel_size', type=int, default=0, 
                        help='VLLM tensor parallel size (0 for auto)')
    parser.add_argument('--gpu_memory_utilization', type=float, default=0.9,
                        help='VLLM GPU memory utilization')
    
    args = parser.parse_args()
    
    # Initialize
    chatbot = ChatBot(
        model_path=args.model_path,
        backend=args.backend,
        temperature=args.temperature,
        max_tokens=args.max_tokens,
        top_p=args.top_p,
        top_k=args.top_k,
        repetition_penalty=args.repetition_penalty,
        dtype=args.dtype,
        device_map=args.device_map,
        tensor_parallel_size=args.tensor_parallel_size,
        gpu_memory_utilization=args.gpu_memory_utilization,
    )
    
    # Interactive mode
    interactive_mode(chatbot)


if __name__ == "__main__":
    main()
```
